src/shell/grabs/moving.rs

```diff
@@ -560,6 +673,22 @@ impl Drop for MoveGrab {
                                 grab_state.window,
                                 window_location.to_local(&workspace.output),
                             );
+
+                            if let Some(sz) = grab_state.snapping_zone {
+                                if sz == SnappingZone::TopMaxim {
+                                    state.common.shell.maximize_toggle(&window, &seat);
+                                } else {
+                                    let direction = match sz {
+                                        SnappingZone::TopSnap => Direction::Up,
+                                        SnappingZone::Bottom => Direction::Down,
+                                        SnappingZone::Left => Direction::Left,
+                                        SnappingZone::Right => Direction::Right,
+                                        _ => Direction::Up,
+                                    };
+                                    window.handle_move(direction);
```

Comment:
Just calling that on the window, won't do what you want. That handles moves e.g. inside a stack.

You want to split the `FloatingLayout::move_current_element` into `FloatingLayout::move_element` and `move_current_element`.

Reply:
fixed in bee0b664cc83aec19f5da5100570c8f17af3c987